cpu/xmc1000/periph/rtc.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (C) 2015 Sebastian Sontberg <sebastian@sontberg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_xmc1000
+ * @{
+ *
+ * @file
+ * @brief       Low-level RTC driver implementation
+ *
+ * @author      Sebastian Sontberg <sebastian@sontberg.de>
+ *
+ * @}
+ */
+
+#include <time.h>
+
+#include "periph/gating.h"
+#include "periph/rtc.h"
+#include "periph_conf.h"
+#include "sched.h"
+#include "thread.h"
+
+/* guard file in case no RTC device was specified */
+#if RTC_NUMOF
+
+/* RTC runs of the internal 32.768kHz standby oscillator */
+#define RTC_DIVIDER (0x7fff)
+
+/* Wait for mirror transfers for this register to finish */
+#define RTC_WAIT(Reg)                                   \
+    do {} while (SCU_GENERAL->MIRRSTS &                 \
+                 SCU_GENERAL_MIRRSTS_RTC_##Reg##_Msk)
+
+/* Storage for callback function and argument */
+static rtc_alarm_cb_t callback;
+static void *callback_arg;
+
+static inline void _rtc_disable(void)
+{
+    RTC_WAIT(CTR);
+    RTC->CTR = (RTC_DIVIDER << RTC_CTR_DIV_Pos);
```

Comment:
Without reading the specs, but wouldn't `RTC->CTR &= ~(1 << RTC_CTR_ENB_Pos)` be less obfuscating?